services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/MongoDbBasedDao.java

```diff
@@ -101,6 +105,25 @@ public final void close() {
         }
     }
 
+    /**
+     * Checks if the given error is caused due to duplicate keys.
+     *
+     * @param error The error to check.
+     * @return {@code true} if the given error is caused by duplicate keys.
+     * @throws NullPointerException if the error is {@code null}.
+     */
+    protected static boolean isDuplicateKeyError(final Throwable error) {
```

Comment:
By moving these methods to this class, `MongoDbDeviceRegistryUtils` is no more used and we can remove it.

Reply:
Absolutely. I thought I already did ...